### PR TITLE
Elixir: Use either local build server or system binary

### PR DESCRIPTION
**Pull request by easbarbosa**
Hi,

Objective:  
Use local binary if found or system one.

Changes: 
- check if there is a local binary in `{EMACS}/.cache/lsp/elixir-ls/language-server.sh` and set command to it.
- if no local binary is found, use system one.

### CODE REVIEW

**Comment by yyoncho** [comment]
@axelson - willing to take a look?

File: clients/lsp-elixir.el [hunk 1]
```diff
@@ -105,6 +105,21 @@ Leave as default to let `executable-find' search for it."
   :type '(repeat string)
   :package-version '(lsp-mode . "7.1"))
 
+
+(defconst lsp-elixir-ls-server-dir
+  (f-join lsp-server-install-dir "elixir-ls")
+  "Elixir-ls local server Directory")
+
+(defcustom lsp-elixir-local-server-command
+  (let ((file (f-join lsp-elixir-ls-server-dir
```
Comment (kiennq) [inline clients/lsp-elixir.el:114]: This checks for file existence whenever lsp's clients are loaded, which make `lsp-mode` loads slower when user is not using `lsp-elixir`.
Can you move this default value to somewhere around here?
``` elisp
                                     `(,(or lsp-elixir-local-server-command
                                            (or (executable-find
                                                 (cl-first lsp-elixir-server-command))
                                                (lsp-package-path 'elixir-ls)))
```

Comment (easbarbosa) [reply clients/lsp-elixir.el:114]: oh, my bad I forgot about that.

**Review by axelson — APPROVED** [review]
This change makes sense to me. I didn't take a close look at the elisp, but the described algorithm sounds correct.

**Comment by easbarbosa** [comment]
@axelson 

I understand that `.cache` are for transient files and `.local` should be better suited for local builds.

Nonetheless, Guix/Nix are an even better solution.

**Review by yyoncho — APPROVED** [review]

**Comment by yyoncho** [comment]
Thank you, sorry for the delay.

**Pull request merged**